### PR TITLE
Added list kuviz endpoint

### PR DESCRIPTION
**Pull request by ethervoid**
Related #14903

### CODE REVIEW

File: app/controllers/carto/api/public/custom_visualizations_controller.rb [hunk 1]
```diff
@@ -1,12 +1,40 @@
...
 
   ssl_required
 
   before_action :validate_input_data, only: [:create, :update]
 
   def index
-    head 501
+    offdatabase_orders = Carto::VisualizationQueryOrderer::SUPPORTED_OFFDATABASE_ORDERS.map(&:to_sym)
+    valid_order_combinations = VALID_ORDER_PARAMS - offdatabase_orders
+    opts = { valid_order_combinations: valid_order_combinations }
```
Comment (gonzaloriestra) [inline app/controllers/carto/api/public/custom_visualizations_controller.rb]: We are not allowing offdatabase params here, so I think these first 3 lines could be replaced with:
```
opts = { valid_order_combinations: VALID_ORDER_PARAMS }
```

File: spec/requests/carto/api/custom_visualizations_controller_spec.rb [hunk 2]
```diff
@@ -59,55 +58,155 @@
...
+                                                       resource: StringIO.new('<html><body>test</body></html>'))
+      @asset_password.save
+    end
+
+    after(:each) do
+      @kuviz.destroy
+      @kuviz_password.destroy
+    end
+
+    it 'shows one visualization' do
```
Comment (gonzaloriestra) [inline spec/requests/carto/api/custom_visualizations_controller_spec.rb]: one?

File: spec/requests/carto/api/custom_visualizations_controller_spec.rb [hunk 3]
```diff
@@ -59,55 +58,155 @@
...
+
+    it 'should return one visualization but total should be two' do
+      get_json api_v4_kuviz_list_vizs_url(api_key: @user.api_key), per_page: 1 do |response|
+        expect(response.status).to eq(200)
+        expect(response.body[:visualizations].size).to eq(1)
+        expect(response.body[:total_entries]).to eq(2)
+      end
+    end
+
+    it 'should order results by legal order fields: name, updated_at and privacy' do
```
Comment (gonzaloriestra) [inline spec/requests/carto/api/custom_visualizations_controller_spec.rb]: Would you mind to split all these calls into different tests?

Comment (ethervoid) [reply spec/requests/carto/api/custom_visualizations_controller_spec.rb]: I rather prefer not to. Those are the same test performed over different parameters. I dont see the point to make one test per parameter, I tend to group them all togheter

File: spec/requests/carto/api/custom_visualizations_controller_spec.rb (hunk 2 above)
Comment (ethervoid) [reply spec/requests/carto/api/custom_visualizations_controller_spec.rb]: Hahaha yes, at first it was one, then I changed it to show everything. Fixed

File: app/controllers/carto/api/public/custom_visualizations_controller.rb (hunk 1 above)
Comment (ethervoid) [reply app/controllers/carto/api/public/custom_visualizations_controller.rb]: Fixed. I thought it was another thing and I didn't checked it but I checked the page_per_page...part :man_facepalming: Good catch :)

File: spec/requests/carto/api/custom_visualizations_controller_spec.rb (hunk 3 above)
Comment (gonzaloriestra) [reply spec/requests/carto/api/custom_visualizations_controller_spec.rb]: I belong to the `one expectation per test` school: I think it's clearer and easier to debug. But ok :)

Comment (ethervoid) [reply spec/requests/carto/api/custom_visualizations_controller_spec.rb]: I'm too but In cases where you're testing one behavior with multiple options sometimes is cleaner and clearer to group it in one test

**Pull request merged**